drivers/pcie/host/pcie.c

```diff
@@ -68,6 +68,83 @@ uintptr_t pcie_get_mbar(pcie_bdf_t bdf, unsigned int index)
 	return PCIE_CONF_BAR_ADDR(addr);
 }
 
+/* The first bit is used to indicate whether the list of reserved interrupts
+ * have been initialized based on content stored in the irq_alloc linker
+ * section in ROM.
+ */
+#define IRQ_LIST_INITIALIZED 0
+
+static ATOMIC_DEFINE(irq_reserved, CONFIG_MAX_IRQ_LINES);
```

Comment:
Why atomic_t and not just integers?  Seems like this is only mutated from initialization code that could easily be locked.  Though I guess it's not much savings in code size as the atomic bit API exists.

Reply:
It was mainly not to reimplement bit manipulation APIs. ATOMIC_DEFINE seemed also convenient since it takes a number of needed bits and gives you a sufficiently large array. But you’re right that it’s perhaps a bit overkill since access is limited to initialization code.